bundler/spec/dependabot/bundler/update_checker/latest_version_finder_spec.rb

```diff
@@ -318,6 +321,39 @@
         end
       end
 
+      context "that we don't have authentication details for", :bundler_v2_only do
+        let(:error_message) do
+          <<~ERR
+            Bad username or password for https://SECRET_CODES@repo.fury.io/greysteil/.
+            Please double-check your credentials and correct them.
+          ERR
+        end
+
+        let(:error_class) do
+          "Bundler::Fetcher::BadAuthenticationError"
```

Comment:
We'll also need to handle it the new error here https://github.com/dependabot/dependabot-core/blob/cbd62f2611af242457227d49add1f1a88ec38542/bundler/lib/dependabot/bundler/update_checker/shared_bundler_helpers.rb#L130 but that's probably integration tested in the update checker so maybe ok to punt on that change here?

Reply:
Good point, let me double check as I think my mocking is off.

Reply:
Ah, we handle this error already in that helper
https://github.com/dependabot/dependabot-core/blob/cbd62f2611af242457227d49add1f1a88ec38542/bundler/lib/dependabot/bundler/update_checker/shared_bundler_helpers.rb#L134

The diff between bundler 1 and bundler 2 is 'creds are missing' is now just the same as 'creds are bad', so we get the behaviour different handled for free with a very slight change in `error_details[:source]` for bundler 2 that is non-breaking.

Reply:
Yes totally missed that, nice one 👍